tools/approval-service/internal/service/ttleventcache_test.go

```diff
@@ -0,0 +1,280 @@
+/*
+ *  Copyright 2025 Gravitational, Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package service
+
+import (
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"testing/synctest"
+	"time"
+)
+
+func (c *TTLEventCache) checkInternalState(t *testing.T, expectedMapLen, expectedListLen int) {
+	t.Helper()
+	c.mu.Lock()
+	defer c.mu.Unlock()
```

Comment:
This locking should not be needed. If it is, then the test is non-deterministic as it is obviously changing underneath the test check, meaning it is a flaky test. Instead if you do not take the lock the race detector should identify this as a problem if there is any concurrency of note that affects the tests.